servers/zts/src/main/java/com/yahoo/athenz/zts/store/impl/S3ChangeLogStore.java

```diff
@@ -71,24 +76,32 @@ public boolean supportsFullRefresh() {
     
     @Override
     public SignedDomain getSignedDomain(String domainName) {
-        
-        // make sure we have an aws s3 client for our request
-        
-        if (awsS3Client == null) {
-            awsS3Client = getS3Client();
-        }
-        
-        SignedDomain signedDomain = getSignedDomain(awsS3Client, domainName);
 
-        // if we got a failure for any reason, we're going
-        // get a new aws s3 client and try again
-        
+        SignedDomain signedDomain = tempSignedDomainMap.get(domainName);
```

Comment:
Here we can just call tempSignedDomainMap.remove(domainName);

remove call returns the existing object so then later on you don't need to call explicitly remove again.